contracts/TellorFlex.sol

```diff
@@ -580,60 +704,112 @@ contract TellorFlex {
     // *                                                                           *
     // *****************************************************************************
 
+    function _updateStakeAmount() internal {
+        (bool valFound, bytes memory val, uint256 timestamp) = getDataBefore(
+            trbUsdSpotPriceQueryId,
+            block.timestamp - 12 hours
+        );
+        if (valFound) {
+            uint256 priceTRB = abi.decode(val, (uint256));
+            stakeAmount = (stakeAmountDollarTarget / priceTRB) * 10**18;
```

Comment:
this should be either `stakeAmount = stakeAmountDollarTarget * 10e18 / priceTRB;`

or we can just make stakeAmountDollarTarget == realDollarTarget * 10e36 and save a little on gas